src/shared/cloudformation/templateSymbolResolver.ts

```diff
@@ -27,34 +31,70 @@ export interface TemplateSymbolProvider {
 export class TemplateSymbolResolver {
     public constructor(
         private readonly document: vscode.TextDocument,
+        // YAML symbol provider (we registered *.yaml in activateCodeLensProviders()).
         private readonly symbolProvider: TemplateSymbolProvider = new DefaultSymbolProvider()
     ) {}
 
     /**
-     * Extracts Function resources from the document.
+     * Extracts Function or implicit Api resources from the document.
+     *
+     * - Function (no Api): `AWS::Serverless::Function` _without_ `Events.Type=Api`.
+     * - Implicit Api: `AWS::Serverless::Function` with `Events.Type=Api`.
```

Comment:
Does this work? `Type: Api` looks like it has to be tied to an `AWS::Serverless::Api`: https://github.com/aws/serverless-application-model/blob/master/versions/2016-10-31.md#api

Reply:
The "implicit case" is what this PR supports, i.e. from the spec:

> An AWS::Serverless::Api resource need not be explicitly added to a AWS Serverless Application Model template. A resource of this type is implicitly created from the union of Api events defined on AWS::Serverless::Function resources defined in the template that do not refer to an AWS::Serverless::Api resource

Top-level `AWS::Serverless::Api` is not supported in this PR.